Orbitersdk/samples/ProjectApollo/src_csm/csm_telecom.cpp

```diff
@@ -1010,23 +1011,29 @@ void HGA::TimeStep(double simt, double simdt)
 
 	//sprintf(oapiDebugString(), "Alpha: %lf� Gamma: %lf� PitchRes: %lf� YawRes: %lf�", Alpha*DEG, Gamma*DEG, PitchRes*DEG, YawRes*DEG);
 
-	VECTOR3 U_RP, pos, R_E, R_M, U_R, U_Earth, U_Moon, U_CSM;
+	VECTOR3 U_RP, pos, R_E = _V(0, 0, 0), R_M, U_R, U_Earth, U_Moon, U_CSM;
 	MATRIX3 Rot;
 	double relang, beamwidth, Moonrelang, EarthSignalDist, CSMrelang;
 
 	OBJHANDLE hMoon = oapiGetObjectByName("Moon");
 	OBJHANDLE hEarth = oapiGetObjectByName("Earth");
+	VESSEL4* MCCVessel = (VESSEL4*)oapiGetVesselInterface(oapiGetVesselByName("MCC"));
 
-	//Global position of Earth, Moon and spacecraft, spacecraft rotation matrix from local to global
+	//Global position of the spacecraft, spacecraft rotation matrix from local to global
 	sat->GetGlobalPos(pos);
-	oapiGetGlobalPos(hEarth, &R_E);
-	oapiGetGlobalPos(hMoon, &R_M);
 	sat->GetRotationMatrix(Rot);
+
+	//Get the gain, power and global position of the transmitter
+	MCCVessel->clbkGeneric(paCBGmessageID::messageID::RF_PROPERTIES, paCBGmessageID::parameterID::GetTxPosition, &R_E);
+	MCCVessel->clbkGeneric(paCBGmessageID::messageID::RF_PROPERTIES, paCBGmessageID::parameterID::GetTxPower, &TransmitterPower);
+	MCCVessel->clbkGeneric(paCBGmessageID::messageID::RF_PROPERTIES, paCBGmessageID::parameterID::GetTxGain, &TransmitterGain);
```

Comment:
Maybe this could be made more efficient by putting all three of these values in a struct that the void point is referencing, so sending this data as a package instead of three separate clbkGeneric calls. Just an idea though.

Reply:
I think I had something similar when I was using connectors. Rf_properties or something like that. Good idea.